source/server/listener_manager_impl.cc

```diff
@@ -643,7 +643,7 @@ std::vector<std::reference_wrapper<Network::ListenerConfig>> ListenerManagerImpl
   std::vector<std::reference_wrapper<Network::ListenerConfig>> ret;
   ret.reserve(active_listeners_.size());
   for (const auto& listener : active_listeners_) {
-    ret.push_back(*listener);
+    ret.emplace_back(*listener);
```

Comment:
Is there a reason why emplace_back is used instead of push_back? https://abseil.io/tips/112 suggests unless you have a specific reason to use emplace_back, push_back is generally preferred for the reasons described there.

Reply:
I assumed that using ```emplace_back``` here helps to avoid additional copy since ```*listener``` is not a prvalue, I will change it back anyway if ```push_back``` is preferred.

Reply:
Yes, it can be argued that emplace_back has a performance edge over push_back. But that comes at the cost of implicit type conversion that may or not may not be safe. I guess the idea is that unless the performance is proven(benchmarked) to be significant, we generally prefer push_back